File: src/game/models/owner.ts
export type OwnerPersonality =
  | 'spendthrift'
  | 'patient'
  | 'win_now'
  | 'frugal'
  | 'meddler'
  | 'hands_off'

export interface OwnerProfile {
  teamId: string
  name: string
  personality: OwnerPersonality
  netWorth: number
  cash: number
  softCashPressureSeasons: number
}

File: scripts/generate/types/owner.go
package types

type OwnerPersonality string

const (
	OwnerSpendthrift OwnerPersonality = "spendthrift"
	OwnerPatient     OwnerPersonality = "patient"
	OwnerWinNow      OwnerPersonality = "win_now"
	OwnerFrugal      OwnerPersonality = "frugal"
	OwnerMeddler     OwnerPersonality = "meddler"
	OwnerHandsOff    OwnerPersonality = "hands_off"
)

type OwnerProfile struct {
	TeamId                 string           `json:"teamId"`
	Name                   string           `json:"name"`
	Personality            OwnerPersonality `json:"personality"`
	NetWorth               float64          `json:"netWorth"`
	Cash                   float64          `json:"cash"`
	SoftCashPressureSeasons int             `json:"softCashPressureSeasons"`
}
